visualization/app/codeCharta/util/fileExtensionCalculator.ts

```diff
@@ -90,18 +79,19 @@ export class FileExtensionCalculator {
 		return {
 			fileExtension,
 			absoluteMetricValue: metricValue,
-			relativeMetricValue: null,
+			relativeMetricValue: 0,
 			color: null
 		}
 	}
 
-	private static getSumOfAllMetrics(distribution: MetricDistribution[]) {
-		return distribution.reduce((partialSum, a) => partialSum + a.absoluteMetricValue, 0)
-	}
-
 	static estimateFileExtension(fileName: string) {
-		if (fileName.includes(".")) {
-			return fileName.split(".").reverse()[0].toLowerCase()
+		const lastDotPosition = fileName.lastIndexOf(".")
+		if (lastDotPosition > 0) {
+			const extension = fileName.slice(lastDotPosition + 1)
+			// TODO: Check Windows path as well?
+			if (!extension.includes("/")) {
+				return extension.toLowerCase()
+			}
```

Comment:
I don't really understand this path. Can't we just return the lowerCase-extension here? Why would we check for a slash in the extension?

Reply:
You are probably correct. I thought about having a full path and not only a filename but that's not the case here. I'll fix that :-)